### ISSUE TITLE
[Feature] Allow roles to be specified when creating API Tokens

### ISSUE THREAD

**Issue by iangabrielsanchez** [nocodb/nocodb#2707]
**Need for this feature**
It allows people to specify permissions to api tokens

**Is your feature request related to a problem? Please describe.**
We are using nocodb APIs to manage custom tables that customers use. When using API Tokens, I noticed that I cannot create tables with them

**Describe alternatives you've considered**
I can create a machine user and use sign in and use the auth token until it expires, then log in again. Is there a way to refresh this token?

**Comment by o1lab** [comment]
I remember, there was a discussion where user didn't want to this. And we made a change to the behaviour. We will consider this.

@vijayrathore8492 : is this also your usecase hence you wanted to have a long expiry in JWT ?

**Comment by vijayrathore8492** [comment]
Not really, We are using JWT token for accessing APIs and we do not have a mechanism in place to refresh the token. So if the token expires, everything will just start failing.

**Comment by o1lab** [comment]
@vijayrathore8492 : no I meant to clarify, are you using tokens to create tables (metadata) ?

**Comment by oyale** [comment]
Related to this, but with the use case on the opposite side, is there a way to create read-only API/auth tokens?

Do I have to create a read-only user and log in with it in order to create read-only access for the REST API?

**Comment by dstala** [comment]
related to https://github.com/nocodb/nocodb/issues/1902

**Comment by alirezamdp** [comment]
Hello everyone, they are very useful if I can give users a token that only has read access.

**Comment by intellent** [comment]
We want to use Noco’s APIs from the frontend of our websites. So the API token would be public. The token should only be allowed to add records to a specific table. No read, update or delete access. That’s not possible, is it?

**Comment by dstala** [comment]
https://nocodb.com/docs/product-docs/account-settings/api-tokens#create-a-fine-grained-token
